src/librustc_typeck/diagnostics.rs

```diff
@@ -236,6 +236,52 @@ fn main() {
 ```
 "##,
 
+E0044: r##"
+You can't use type parameters on foreign items. Example of erroneous code:
+
+```
+extern { fn some_func<T>(); }
+```
+
+Just remove the type parameter to make this code works:
+
+```
+extern { fn some_func(); }
+```
+"##,
+
+E0045: r##"
```

Comment:
https://github.com/rust-lang/rust/blob/40d19bf64107622763f9b2c1a94e1c7576138b14/src/librustc_typeck/diagnostics.rs#L1470

E0045 is about non-foreign functions with variadic params